### ISSUE TITLE
Configure jest to perform es6 module transforms during testing

### ISSUE THREAD

**Issue by dougajmcdonald** [leebenson/reactql#96]
I'm trying to export an es6 module and import it into a test but I'm having some trouble getting the jest config to play nicely with es6 modules.

Do you know what I'd need to configure to allow the es6 transpile to work correctly? It looks like there is some `jest.transform.js` bits and bobs but they don't seem to operate for me. 
 
**src/sum.js**
```
const sum = (a, b) => a + b;

export default sum;
```

**src/sum.test.js**
```
import sum from 'src/sum';

test('adds 1 + 2 to equal 3', () => {
  expect(sum(1, 2)).toBe(3);
});
```

`npm test` output:

```
> @2.10.0 test c:\github\gareththegeek\corewar-reactql
> jest

 FAIL  src\sum.test.js
  ● Test suite failed to run

    c:\github\gareththegeek\corewar-reactql\src\sum.test.js:2
    import sum from 'src/sum';
    ^^^^^^

    SyntaxError: Unexpected token import

      at ScriptTransformer._transformAndBuildScript (node_modules/jest-runtime/build/script_transformer.js:305:17)
          at Generator.next (<anonymous>)
          at new Promise (<anonymous>)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        1.757s
Ran all test suites.
```

**Comment by biels** [comment]
This is a deal breaker issue for me. I hope it gets addressed soon.

**Comment by dougajmcdonald** [comment]
I found the problem, it's a simple one, the regex is wrong!

the `(x)` isn't an optional x which I presume is what it's meant to be.

in your `jest.config.js` change the `transform` bit from:

```
  transform: {
    '^.+\\.js(x)$': '<rootDir>/jest.transform.js',
  },
```

to

```
  transform: {
    '^.+\\.jsx?$': '<rootDir>/jest.transform.js',
  },
```

The `?` matches 0-1 preceding characters so is the right one to use here.

**Comment by leebenson** [comment]
There's actually no test runner included in the 3.0.0 kit (released today) - I took it out for speed of the new release, since I wasn't settled on a testing approach yet.

I'll look at this again soon. If it's required in your project, a test runner can easily be added.
